src/renderer/components/screens/AccountListSidebar/AccountListSidebar.tsx

```diff
@@ -0,0 +1,119 @@
+import { T } from '@deltachat/jsonrpc-client'
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { BackendRemote } from '../../../backend-com'
+import { AccountItem } from './AccountItem'
+
+import styles from './styles.module.scss'
+import { AccountHoverInfo } from './AccountHoverInfo'
+import { debounce } from 'debounce'
+import { runtime } from '../../../runtime'
+
+export function AccountListSidebar({
+  selectedAccountId,
+  onAddAccount,
+  onSelectAccount,
+  openAccountDeletionScreen,
+}: {
+  selectedAccountId: number | undefined
+  onAddAccount: () => Promise<number>
+  onSelectAccount: (accountId: number) => Promise<void>
+  openAccountDeletionScreen: (accountId: number) => Promise<void>
+}) {
+  const [accounts, setAccounts] = useState<T.Account[]>([])
+
+  const selectAccount = async (accountId: number) => {
+    if (selectedAccountId === accountId) {
+      return
+    }
+
+    await onSelectAccount(accountId)
+  }
+
+  const [syncAllAccounts, setSyncAllAccounts] = useState(true)
+
+  const refresh = useMemo(
+    () => async () => {
+      const accounts = await BackendRemote.rpc.getAllAccounts()
+      setAccounts(accounts)
+      const desktopSettings = await runtime.getDesktopSettings()
+      setSyncAllAccounts(desktopSettings.syncAllAccounts)
+    },
+    []
+  )
+
+  useEffect(() => {
+    refresh()
+  }, [selectedAccountId, refresh])
+
+  const [accountForHoverInfo, internalSetAccountForHoverInfo] =
+    useState<T.Account | null>(null)
+
+  const updateAccountForHoverInfo = (
+    actingAccount: T.Account,
+    select: boolean
+  ) => {
+    internalSetAccountForHoverInfo(oldAccount => {
+      if (actingAccount === oldAccount && select === false) {
+        // only deselect if it is really deselecting the current one
+        return null
+      }
+      if (select) return actingAccount
+      return null
+    })
+  }
+
+  const hoverInfo = useRef<HTMLDivElement | null>(null)
+
+  const updateHoverInfoPosition = useCallback(() => {
+    if (hoverInfo.current && accountForHoverInfo) {
+      const elem = document.querySelector(
+        `div[x-account-sidebar-account-id="${accountForHoverInfo.id}"]`
+      )
+      if (elem) {
+        const rect = elem.getBoundingClientRect()
+        hoverInfo.current.style.top = `${rect.top}px`
+        hoverInfo.current.style.left = `${rect.right + 15}px`
+      }
+    }
+  }, [accountForHoverInfo])
+
+  useEffect(() => {
+    updateHoverInfoPosition()
+  }, [accountForHoverInfo, updateHoverInfoPosition])
+
+  useEffect(() => {
+    window.__updateAccountListSidebar = debounce(() => {
```

Comment:
We should try to avoid using `window`, it creates global state which gets hard to understand and to debug. For this I'd use contexts where it is clear where the state lives and who it consumes.

Reply:
Oh, I see, this is deeply intertwined with the settings store .. okay, let's leave it then as a hack here as I can see how a refactor is too much work

Reply:
I added it to the todo in https://github.com/deltachat/deltachat-desktop/issues/3671